pkg/resolver/file/file.go

```diff
@@ -142,135 +200,276 @@ func (r *Resolver) yamlWalk(value *yaml.Node, path string, resolveCount int, res
 }
 
 // isPath returns true if the value is a valid path
-func (r *Resolver) resolveYamlPath(v *yaml.Node, filePath string,
-	resolveCount int, resolvedFilesCache map[string]ResolvedFile) (yaml.Node, bool) {
+func (r *Resolver) resolveYamlPath(
+	originalFileContent []byte,
+	fullObject *yaml.Node,
+	v *yaml.Node,
+	filePath string,
+	resolveCount int,
+	resolvedFilesCache map[string]ResolvedFile) (yaml.Node, bool) {
 	value := v.Value
 	if resolveCount > constants.MaxResolvedFiles {
 		return *v, false
 	}
+	var splitPath []string
+	var obj *yaml.Node
+	sameFileResolve := false
+	if strings.HasPrefix(value, "#") { // same file resolve
+		sameFileResolve = true
+		path := filePath + value
+		splitPath = strings.Split(path, "#") // splitting by removing the section to look for in the file
+		obj = fullObject
+	} else { // external file resolve
+		value = checkServerlessFileReference(value)
+
+		path := filepath.Join(filepath.Dir(filePath), value)
+		splitPath = strings.Split(path, "#") // splitting by removing the section to look for in the file
+		onlyFilePath := splitPath[0]
+		_, err := os.Stat(path)
+		if err != nil {
+			return *v, false
```

Comment:
```suggestion
                // if path is not valid or not exists return original value 
		onlyFilePath := splitPath[0]
		_, err := os.Stat(path)
		if err != nil {
			return *v, false
```

Add comments for the returns expressing what is returned and why

Reply:
Done here: https://github.com/Checkmarx/kics/pull/6405/commits/b36e81021f208c1e2ee9015f034e6cc69be9460b